src/snowcli/cli/nativeapp/init.py

```diff
@@ -0,0 +1,291 @@
+from pathlib import Path
+import subprocess
+from tempfile import TemporaryDirectory
+from click.exceptions import ClickException
+
+
+from typing import Optional
+from snowcli.cli.project.definition import DEFAULT_USERNAME
+from snowcli.cli.project.util import clean_identifier, get_env_username
+from snowcli.cli.render.commands import generic_render_template
+
+from snowcli.utils import get_client_git_version
+
+
+log = logging.getLogger(__name__)
+
+SNOWFLAKELABS_GITHUB_URL = "https://github.com/Snowflake-Labs/native-app-templates"
+BASIC_TEMPLATE = "native-app-basic"
+
+
+class InitError(ClickException):
+    """
+    Native app project could not be initiated due to an underlying error.
+    """
+
+    def __init__(self):
+        super().__init__(self.__doc__)
+
+
+class GitVersionIncompatibleError(ClickException):
+    """
+    Init requires git version to be at least 2.25.0. Please update git and try again.
+    """
+
+    def __init__(self):
+        super().__init__(self.__doc__)
+
+
+class GitCloneError(ClickException):
+    """
+    Could not complete git clone with the specified git repository URL.
+    """
+
+    def __init__(self):
+        super().__init__(self.__doc__)
+
+
+class RenderingFromJinjaError(ClickException):
+    """
+    Could not complete rendering file from Jinja template.
+    """
+
+    # def __init__(self, message: str):
+    #     super.__init__(message)
+    def __init__(self, name: str):
+        super().__init__(
+            f"Could not complete rendering file from Jinja template: {name}"
+        )
+
+
+class CannotInitializeAnExistingProjectError(ClickException):
+    """
+    Cannot initialize a new project within an existing Native Application project.
+    """
+
+    def __init__(self):
+        super().__init__(self.__doc__)
+
+
+class DirectoryAlreadyExistsError(ClickException):
+    """
+    Directory already contains a project with the intended name
+    """
+
+    name: str
+
+    def __init__(self, name: str):
+        super().__init__(
+            f"This directory already contains a sub-directory called {name}. Please try a different name."
+        )
+        self.name = name
+
+
+def _sparse_checkout(
+    git_url: str, repo_sub_directory: str, target_parent_directory: str
+):
+    """
+    Clone the requested sub directory of a git repository from the provided git url.
+
+    Args:
+        git_url (str): The full git url to the repository to be cloned.
+        repo_sub_directory (str): The sub directory name within the repository to be cloned.
+        target_parent_directory (str): The parent directory where the git repository will be cloned into.
+
+    Returns:
+        None
+    """
+
+    clone_command = (
+        f"git clone -n --depth=1 --filter=tree:0 {git_url} {target_parent_directory}"
+    )
+    sparse_checkout_command = f"""
+        cd {target_parent_directory} &&
+            git sparse-checkout set --no-cone {repo_sub_directory} &&
+                git checkout
+        """
```

Comment:
As we added Git Python as dependency, shouldn't we use https://gitpython.readthedocs.io/en/stable/reference.html#git.repo.base.Repo.clone?